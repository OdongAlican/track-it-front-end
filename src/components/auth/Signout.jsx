import React from 'react';
import { useDispatch } from 'react-redux';
import { signOut } from '../../actions/index';

const Signout = () => {
  const dispatch = useDispatch();

  const logOut = () => {
    dispatch(signOut());
  };

  return (
    <div data-testid="appSignout">
      <button
        onClick={logOut}
        className="btn btn-secondary"
        type="button"
      >
        Sign Out
      </button>
    </div>
  );
};

export default Signout;
