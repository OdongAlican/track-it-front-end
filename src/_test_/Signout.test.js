import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Signout from '../components/auth/Signout';

afterEach(cleanup);
afterEach(cleanup);
const initialState = {
  authReducer: {
    output: 10,
    error: 'test',
  },
};
const mockStore = configureStore();
let store;

it('renders Signout correctly', () => {
  store = mockStore(initialState);
  const { getAllByTestId } = render(
    <Router>
      <Provider store={store}><Signout /></Provider>
      ,
    </Router>,
  );
  const result = getAllByTestId('appSignout');
  expect(result[0]).toHaveTextContent('Sign Out');
});

it('renders Signout correctly', () => {
  store = mockStore(initialState);
  const { getAllByTestId } = render(
    <Router>
      <Provider store={store}><Signout /></Provider>
      ,
    </Router>,
  );
  const result = getAllByTestId('appSignout');
  expect(result[0]).not.toHaveTextContent('Some Random');
});

it('matches Signout snapshot', () => {
  const appointment = renderer.create(
    <Router>
      <Provider store={store}><Signout /></Provider>
    </Router>,
  );
  const tree = appointment.toJSON();
  expect(tree).toMatchSnapshot();
});
