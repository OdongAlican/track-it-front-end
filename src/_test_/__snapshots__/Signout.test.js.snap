// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Signout snapshot 1`] = `
<div
  data-testid="appSignout"
>
  <button
    className="btn btn-secondary"
    onClick={[Function]}
    type="button"
  >
    Sign Out
  </button>
</div>
`;
